protocols/relay/examples/Dockerfile

```diff
@@ -0,0 +1,22 @@
+# 1: Build the exe
+FROM rust:latest as builder
+LABEL Name=libp2p-relay Version=0.0.1
+# 1a: Prepare for static linking
+RUN apt-get update && \
+    apt-get dist-upgrade -y && \
+    apt-get install -y musl-tools && \
+    rustup target add x86_64-unknown-linux-musl
+# 1b: Download and compile Rust dependencies (and store as a separate Docker layer)
+RUN USER=root cargo new rust-libp2p
+WORKDIR /usr/src/libp2p
+
+COPY . /usr/src/libp2p
+RUN cargo build --target x86_64-unknown-linux-musl --example=relay --package=libp2p-relay
```

Comment:
Why is the copying required here? Afterall this is the build stage, right?

Reply:
Without this copy the cargo build will not work. It should refer the entire code base of libp2p.
(This is not the COPY of the build artifacts)
Can you provide other suggestion?

Reply:
As far as I can tell the `RUN` step is not needed:

```Dockerfile
# 1: Build the exe
FROM rust:latest as builder
LABEL Name=libp2p-relay Version=0.0.1
# 1a: Prepare for static linking
RUN apt-get update && \
    apt-get dist-upgrade -y && \
    apt-get install -y musl-tools && \
    rustup target add x86_64-unknown-linux-musl
# 1b: Download and compile Rust dependencies (and store as a separate Docker layer)
WORKDIR /usr/src/libp2p

COPY . .
RUN cargo build --target x86_64-unknown-linux-musl --example=relay --package=libp2p-relay

# 2: Copy the exe and extra files ("static") to an empty Docker image
FROM debian:buster-slim

COPY --from=builder /usr/src/libp2p/target/x86_64-unknown-linux-musl/debug/examples/relay .
USER 1000

ENTRYPOINT ./relay -d $deterministic_peer_env --mode $relay_command_arg_env --address $address_env

```